daliuge-engine/dlg/drop.py

```diff
@@ -1715,6 +1717,24 @@ def initialize(self, **kwargs):
         super(BarrierAppDROP, self).initialize(**kwargs)
 
 
+class PlasmaDROP(AbstractDROP):
+    '''
+    A DROP that points to data stored in a Plasma Store
+    '''
+    object_id = dlg_string_param('object_id', plasma.ObjectID.from_random().binary())
+    plasma_link = dlg_string_param('plasma_link', '/tmp/plasma')
+
+    def initialize(self, **kwargs):
+       pass
+
+    def getIO(self):
+        return PlasmaIO(plasma.ObjectID(self.object_id), plasma_link)
+
+    @property
+    def dataURL(self):
+        return "plasma://%s:%d/%s" % (plasma.ObjectID(self.object_id))
```

Comment:
This will fail to format, you'll want to double-check what you want to put into the URL (should make it more-or-less unique), just using the object ID could be enough as a first iteration (and we don't use `dataURL` much anyway).

Reply:
I really was not sure what should go here. `self.object_id` should probably be used instead as it is a `dlg_string_param`.

Reply:
Will use `self.uid` instead.